### PR TITLE
OCPBUGS-24177: fix race for MCN creation

### PR DESCRIPTION
**Pull request by cdoern**
the UpgradeMonitor and the Operator try to create the MCN if they DNE. if the Upgrade Monitor gets there first, it needs to set spec.DesiredConfig to  sane value

### CODE REVIEW

File: pkg/upgrademonitor/upgrade_monitor.go
Comment (sinnykumari) [inline pkg/upgrademonitor/upgrade_monitor.go:201]: Wondering if we should just stick to "" to reflect node annotation value or we are adding this string for better readability for user?

Comment (cdoern) [reply pkg/upgrademonitor/upgrade_monitor.go:201]: the API type has to be len > 0. I think we should change this in 4.16 but the API team insisted on it. So, we are stuck with having some default value.

File: pkg/upgrademonitor/upgrade_monitor.go
Comment (yuqi-zhang) [inline pkg/upgrademonitor/upgrade_monitor.go:171]: non-blocking: can probably switch these to constants from the daemon

**Review by yuqi-zhang — APPROVED** [review]
/lgtm
/hold

for pre-merge verification

File: pkg/upgrademonitor/upgrade_monitor.go
Comment (sinnykumari) [reply pkg/upgrademonitor/upgrade_monitor.go:201]: ah, thanks for the context

**Review by sinnykumari — COMMENTED** [review]
/lgtm
Pending QE testing for verification

**Comment by cdoern** [comment]
@rioliu-rh or @sergiordlr please remove the hold once QE approval has been given. Thanks! We are looking to merge this before FF so that we do not need to backport.

File: pkg/upgrademonitor/upgrade_monitor.go
Comment (cdoern) [reply pkg/upgrademonitor/upgrade_monitor.go:171]: I am going to leave it unless tests fail but I'll mark this down for things to change. Focusing here more on functionality breaking stuff.

**Comment by rioliu-rh** [comment]
@cdoern I still can find below error in MCD log
```console
E1130 04:15:58.848391   15754 upgrade_monitor.go:267] error getting existing MCN: %!w(*errors.StatusError=&{{{ } {   <nil>} Failure machineconfignodes.machineconfiguration.openshift.io "ip-10-0-95-106.us-east-2.compute.internal" not found NotFound 0xc000b44660 404}}) 
```

**Comment by cdoern** [comment]
That is not an error but a log. Just to let us know the race occurs in certain scenarios @rioliu-rh if the other log is gone then this is fixed.

**Pull request merged**
